### PR TITLE
Global app nav mf 430

### PR DESCRIPTION
**Pull request by joachim-junior**
Hi Sir, 
- I have added the Global application nav bar switcher Icon.
- I have equally added an extension slot
- Added a close button to close the nav bar.

### CODE REVIEW

File: src/components/navbar/navbar.component.tsx [hunk 1]
```diff
@@ -94,6 +106,39 @@ const Navbar: React.FC<NavbarProps> = ({
                 >
                   <UserAvatarFilledAlt20 />
                 </HeaderGlobalAction>
+                <HeaderGlobalAction
```
Comment (FlorianRappl) [inline src/components/navbar/navbar.component.tsx:109]: Just use a single one. Also all the labels etc. are not right.

File: src/components/navbar/navbar.component.tsx [hunk 2]
```diff
@@ -94,6 +106,39 @@ const Navbar: React.FC<NavbarProps> = ({
...
+                  aria-label="Add"
+                  aria-labelledby="Add Button"
+                  name="AddNavButton"
+                  style={{ display: switcherToggler ? "none" : "block" }}
+                  onClick={handleSwitcherClick}
+                >
+                  <AppSwitcher20 />
```
Comment (FlorianRappl) [inline src/components/navbar/navbar.component.tsx:116]: This should be displayed according to the state.

File: src/components/navbar/navbar.component.tsx [hunk 3]
```diff
@@ -55,6 +57,16 @@ const Navbar: React.FC<NavbarProps> = ({
     }
   };
 
+  const [switcherToggler, setSwitcherToggler] = React.useState(false);
+
+  const handleSwitcherClick = () => {
```
Comment (FlorianRappl) [inline src/components/navbar/navbar.component.tsx:62]: Wrap the CB in `useCallback` and make it a toggler (`setSwitcherToggle(state => !state)`)

File: src/components/navbar/navbar.component.tsx [hunk 4]
```diff
@@ -94,6 +106,39 @@ const Navbar: React.FC<NavbarProps> = ({
...
+                  onClick={handleCloseClick}
+                >
+                  <Close20 />
+                </HeaderGlobalAction>
+                <HeaderPanel
+                  aria-label="Header Panel"
+                  style={{
+                    display: switcherToggler ? "block" : "none",
+                    height: "12rem",
+                    overFlow: "Hidden"
```
Comment (FlorianRappl) [inline src/components/navbar/navbar.component.tsx:135]: I guess you mean `hidden` here? Also I think its just `overflow`.

File: src/components/navbar/navbar.component.tsx [hunk 5]
```diff
@@ -94,6 +106,39 @@ const Navbar: React.FC<NavbarProps> = ({
...
+                  }}
+                  expanded
+                >
+                  <ExtensionSlot extensionSlotName="global-nav-options"></ExtensionSlot>
```
Comment (FlorianRappl) [inline src/components/navbar/navbar.component.tsx:139]: 1. The convention is to end with `-slot`
2. Not sure about the name. The left side is already called `nav-menu-slot`, so I am not sure what this is. What is this menu called?

**Review by FlorianRappl — CHANGES_REQUESTED** [review]

File: src/components/navbar/navbar.component.tsx (hunk 5 above)
Comment (joachim-junior) [reply src/components/navbar/navbar.component.tsx:139]: Sir, the slot has not yet been created for this.

File: src/components/navbar/navbar.component.tsx (hunk 1 above)
Comment (joachim-junior) [reply src/components/navbar/navbar.component.tsx:109]: Sir, right now I'm implementing the click even on the buttons and not the icons. If I use one button I'll implement the even on the Icons. So Clicking any where on the button asides the icon will not work.

**Pull request closed without merging**